packages/lib/services/commands/stateToWhenClauseContext.ts

```diff
@@ -68,7 +70,7 @@ export default function stateToWhenClauseContext(state: State, options: WhenClau
 
 		// Current location
 		inConflictFolder: state.selectedFolderId === Folder.conflictFolderId(),
-		inTrash: state.selectedFolderId === getTrashFolderId() || commandFolder && !!commandFolder.deleted_time,
+		inTrash: (state.selectedFolderId === getTrashFolderId() || commandFolder && !!commandFolder.deleted_time) && hasDeletedTime,
```

Comment:
There's already a check on `deleted_time` here. Couldn't you just add a check on the selected note?

Reply:
That right, I just kept the check for the `selectedNote` object in this case and included it in the original inTrash boolean logic.